src/Idris/Desugar.idr

```diff
@@ -82,72 +82,92 @@ extendSyn newsyn
            , "New (" ++ unwords (map show $ saveMod newsyn) ++ "): "
               ++ show (modDocstrings newsyn)
            ]
-         -- Check there are no conflicting definitions
-         let oldprefix = map (Prefix,) <$> (prefixes syn)
-         let newprefix = map (Prefix,) <$> (prefixes newsyn)
-         let oldinfix  = infixes syn
-         let newinfix  = infixes newsyn
-
-         let check = mapMaybe (\ entry@(k, ((_, v), (_, v'))) => entry <$ guard (k /= "-" && (v /= v')))
-                   . StringMap.toList
-         let inter = concatMap {t = List} {m = List _} check
-                   [ sharedSupport oldprefix newprefix
-                   , sharedSupport oldprefix newinfix
-                   , sharedSupport oldinfix  newprefix
-                   , sharedSupport oldinfix  newinfix
-                   ]
-         unless (null inter) $ recordWarning $ GenericWarn emptyFC $ unlines
-           $ ("Conflicting fixity declarations:" ::)
-           $ flip (concatMap {m = List _}) inter
-           $ \ (k, ((fc1, fix1, prec1), (fc2, fix2, prec2))) => pure $ unwords
-           [ "\{k}:"
-           , "\{show fix1} \{show prec1} (at \{show fc1})", "and"
-           , "\{show fix2} \{show prec2} (at \{show fc2})"
-           ]
 
-         put Syn ({ infixes $= mergeLeft (infixes newsyn),
-                    prefixes $= mergeLeft (prefixes newsyn),
+         put Syn ({ infixes $= merge (infixes newsyn),
+                    prefixes $= merge (prefixes newsyn),
                     ifaces $= merge (ifaces newsyn),
                     modDocstrings $= mergeLeft (modDocstrings newsyn),
                     modDocexports $= mergeLeft (modDocexports newsyn),
                     defDocstrings $= merge (defDocstrings newsyn),
                     bracketholes $= ((bracketholes newsyn) ++) }
                   syn)
 
+-- Check if a namespace is pointing toward a fixity
+-- An namespace can end in an operator fixity if it's in a %hide
+-- directive in order to remove the fixity from the module in order
+-- to better control parsing in the file.
+isOperatorFixityNS : Name -> Bool
+isOperatorFixityNS (NS ns _) = any (`isInPathOf` ns) fixityKeywords
+isOperatorFixityNS _ = False
+
```

Comment:
I think we should be able to garbage collect that definition now.

```suggestion
```

Reply:
Fixed, I also noticed `checkFixity` wasn't doing anything anymore so it's gone